server/model/monitor.js

```diff
@@ -175,6 +177,43 @@ class Monitor extends BeanModel {
                     bean.ping = await ping(this.hostname);
                     bean.msg = ""
                     bean.status = UP;
+                } else if (this.type === "dns") {
+                    let startTime = dayjs().valueOf();
+                    let dnsMessage = "";
+
+                    let dnsRes = await dnsResolve(this.hostname, this.dns_resolve_server, this.dns_resolve_type);
+                    bean.ping = dayjs().valueOf() - startTime;
+
+                    if (this.dns_resolve_type == "A" || this.dns_resolve_type == "AAAA" || this.dns_resolve_type == "CNAME" || this.dns_resolve_type == "PTR") {
+                        dnsMessage = dnsRes[0];
```

Comment:
Actually this comes back to the problem of multiple A records that I had. Is it a reasonable assumption that the user only wants to see the first result? Or do we want to show all of them? I'm honestly not sure.

Reply:
That's a good thought, i think all them is better, adjusted it in commit [`44d9fa6`](https://github.com/louislam/uptime-kuma/pull/238/commits/44d9fa63f06ec97f72bc258404a48a87c651954c)